test/vfd_swmr_dsetops_writer.c

```diff
@@ -0,0 +1,1772 @@
+                /* Wait for a few ticks for the update to happen */
+                decisleep(config.tick_len * s.update_interval);
+
+                result = verify_dsets_action(SEQ_WRITE, &s, &ds, step);
+
+                if (s.use_np && !np_reader(result, step, &s, &np)) {
+                    printf("np_reader() for verifying addition failed\n");
+                    TEST_ERROR;
+                }
+            }
+        }
+
+        /* Start verifying random writes for contiguous and/or chunked datasets */
+        if ((s.contig || s.chunked) && s.rwrites) {
+            unsigned newstep;
+
+            /* Set up random seed which will be the same for both writer and reader */
+            HDsrandom(RANDOM_SEED);
+
+            for (step = 0; (step < s.rwrites && step < (s.rows * s.cols)); step++) {
+                dbgf(2, "Verify random writes %u to dataset\n", step);
+
+                newstep = HDrandom() % (s.rows * s.cols);
+                printf("Random step is %u\n", newstep);
+
+                if (s.use_np && !np_confirm_verify_notify(np.fd_writer_to_reader, step, &s, &np)) {
+                    printf("np_confirm_verify_notify() verify/notify not in sync failed\n");
+                    TEST_ERROR;
+                }
+
+                /* Wait for a few ticks for the update to happen */
+                decisleep(config.tick_len * s.update_interval);
+
+                result = verify_dsets_action(RANDOM_WRITE, &s, &ds, newstep);
+
+                if (s.use_np && !np_reader(result, step, &s, &np)) {
+                    printf("np_reader() for verifying addition failed\n");
+                    TEST_ERROR;
+                }
+            }
+        }
+
+        /* Start verifying hyperslab writes for contiguous and/or chunked datasets */
+        if ((s.contig || s.chunked) && s.lwrites) {
+            unsigned k;
+
+            for (step = 0, k = 0; (step < s.lwrites && k < (s.rows * s.cols)); step++, k += s.cols) {
+                dbgf(2, "Verify hyperslab writes %u to dataset\n", step);
+
+                if (s.use_np && !np_confirm_verify_notify(np.fd_writer_to_reader, step, &s, &np)) {
+                    printf("np_confirm_verify_notify() verify/notify not in sync failed\n");
+                    TEST_ERROR;
+                }
+
+                /* Wait for a few ticks for the update to happen */
+                decisleep(config.tick_len * s.update_interval);
+
+                result = verify_dsets_action(HYPER_WRITE, &s, &ds, k);
+
+                if (s.use_np && !np_reader(result, step, &s, &np)) {
+                    printf("np_reader() for verifying addition failed\n");
+                    TEST_ERROR;
+                }
+            }
+        }
+
+        /* Start verifying raw data modifications for contiguous and/or chunked datasets */
+        if ((s.contig || s.chunked) && s.wwrites) {
+            unsigned k;
+
+            for (step = 0; (step < s.wwrites && step < (s.rows * s.cols)); step++) {
```

Comment:
If I understand correctly, by default,  s.rows is 256 and s.cols is 512, for the increasing of each step, the named pipe is used to shake hand, will it take quite a long time to finish the test?

Reply:
You are right.  I think I will decrease the default rows and columns to e.g.  10 x 5.